0.10/s2i/bin/assemble

```diff
@@ -47,8 +47,51 @@ if [ -n "$NPM_MIRROR" ]; then
 	npm config set registry $NPM_MIRROR
 fi
 
-echo "---> Building your Node application from source"
-npm install
+# Set the DEV_MODE to false by default.
+if [ -z "$DEV_MODE" ]; then
+  export DEV_MODE=false
+fi
+
+# If NODE_ENV is not set by the user, then NODE_ENV is determined by whether
+# the container is run in development mode.
+if [ -z "$NODE_ENV" ]; then
+  if [ "$DEV_MODE" == true ]; then
+    export NODE_ENV=development
+  else
+    export NODE_ENV=production
+  fi
+fi
+
+if [ "$NODE_ENV" != "production" ]; then
+
+	echo "---> Building your Node application from source"
+	npm install
+
+else
+
+	echo "---> Installing all dependencies"
+	NODE_ENV=development npm install
+
+	#do not fail when there is no build script
+	echo "---> Building in production mode"
+	npm run build --if-present
+
+	echo "---> Pruning the development dependencies"
+	npm prune
+
+	# Clear the npm's cache and tmp directories only if they are not a docker volumes
+	NPM_CACHE=$(npm config get cache)
+	if ! mountpoint $NPM_CACHE; then
+		echo "---> Cleaning the npm cache $NPM_CACHE"
+		npm cache clean
+	fi
+	NPM_TMP=$(npm config get tmp)
+	if ! mountpoint $NPM_TMP; then
+		echo "---> Cleaning the $NPM_TMP/npm-*"
+		rm -rf $NPM_TMP/npm-*
+	fi
+
+fi
```

Comment:
Could the script be reduced to this instead?

```
// Need to install both prod and dev deps (for build)
npm install --only=dev
npm install --only=prod

// Build Babel/TypeScript and run other build logic defined by developers
npm run build --if-present

// Clean cache and tmp dir
{CODE IN PR}

// Prune out devDependencies for production deploys
if [ "$NODE_ENV" == "production" ]; then
  npm rune
fi
```

Reply:
Yes, it could be reduced/improved further but first I need a confirmation from the project owners that this PR is going to get merged. As you could see I'm waiting since November without much feedback.